Automattic/sensei
# Add lesson block status preview

Add status preview control for the Lesson Block.

### Notes

- I set up the icon in CSS because this seemed like the simplest way to make it work consistently in the editor and on the frontend. A possibly option would be to create icon images (SVG's?) and switch between them using CSS (this would be less affected by theme fonts, for example). I tried this, but couldn't navigate the build system well enough to make an icon available from both the editor and the backend. If this is easy, we should consider doing it this way instead.
- I extracted the `StatusControl` component, so if we're happy with it we could use it for the Module status preview as well.
- I haven't given any thought to the consistency between the frontend rendering of this block vs. the frontend rendering of legacy courses. There may be some easy things we can do here (with CSS classes, etc) that would help ease the transition for sites with custom styling.

### Testing instructions

- Ensure the feature flag is set for the Course Outline block.
- Create a course with at least two Lessons.
- Click on a Lesson block. You should see the "Preview Lesson Status" panel in the sidebar.
- Use the sidebar to preview the "In Progress" and "Completed" states.
- Open the course in the frontend and start taking the course. Complete one lesson.
- View the course page. You should see the completion indicator for the completed lesson, and not the incomplete lesson.
